tests/subsys/bluetooth/mesh/light_ctrl/src/main.c

```diff
@@ -0,0 +1,290 @@
+	expect_light_onoff_pub(expected_msg, 1);
+}
+
+static void expect_ctrl_enable(void)
+{
+	expected_transition.time = 0;
+	expected_set.lvl = light_ctrl_srv.cfg.light[LIGHT_CTRL_STATE_STANDBY];
+	expect_transition_start();
+}
+
+static void expect_turn_off_state_change(void)
+{
+	expected_transition.time = light_ctrl_srv.cfg.fade_standby_manual;
+	expected_set.lvl = light_ctrl_srv.cfg.light[LIGHT_CTRL_STATE_STANDBY];
+	expect_transition_start();
+
+	expected_msg[0] = 1;
+	expected_msg[1] = 0;
+	expected_msg[2] = 0;
+	expected_onoff_status.present_on_off = true;
+	expected_onoff_status.target_on_off = false;
+	expected_onoff_status.remaining_time = 510;
+	expect_light_onoff_pub(expected_msg, 3);
+}
+
+static void expect_turn_on_state_change(void)
+{
+	expected_transition.time = light_ctrl_srv.cfg.fade_on;
+	expected_set.lvl = light_ctrl_srv.cfg.light[LIGHT_CTRL_STATE_ON];
+	expect_transition_start();
+
+	expected_msg[0] = 1;
+	expected_msg[1] = 1;
+	expected_msg[2] = 0;
+	expected_onoff_status.present_on_off = true;
+	expected_onoff_status.target_on_off = true;
+	expected_onoff_status.remaining_time = 510;
+	expect_light_onoff_pub(expected_msg, 3);
+}
+
+static void
+expected_statemachine_cond(bool enabled,
+			   enum bt_mesh_light_ctrl_srv_state expected_state,
+			   atomic_t expected_flags)
+{
+	if (enabled) {
+		zassert_not_null(light_ctrl_srv.lightness->ctrl,
+				 "Is not enabled.");
+	} else {
+		zassert_is_null(light_ctrl_srv.lightness->ctrl,
+				"Is not disabled");
+	}
+	zassert_equal(light_ctrl_srv.state, expected_state, "Wrong state");
+	zassert_equal(light_ctrl_srv.flags, expected_flags,
+		      "Wrong Flags: 0x%X:0x%X", light_ctrl_srv.flags,
+		      expected_flags);
+}
+
+static void test_fsm_no_change_by_light_onoff(void)
+{
+	/* OFF state, the light ctrl server is disable after init */
+	enum bt_mesh_light_ctrl_srv_state expected_state =
+		LIGHT_CTRL_STATE_STANDBY;
+	atomic_t expected_flags = FLAGS_CONFIGURATION;
+
+	expected_statemachine_cond(false, expected_state, expected_flags);
+
+	bt_mesh_light_ctrl_srv_on(&light_ctrl_srv);
+	expected_statemachine_cond(false, expected_state, expected_flags);
+
+	bt_mesh_light_ctrl_srv_off(&light_ctrl_srv);
+	expected_statemachine_cond(false, expected_state, expected_flags);
+
+	/* STANDBY state */
+	/* Expecting flags to change */
+	expected_flags = expected_flags | BIT(FLAG_CTRL_SRV_MANUALLY_ENABLED);
+	expected_flags = expected_flags | BIT(FLAG_TRANSITION);
+	expect_ctrl_enable();
+	bt_mesh_light_ctrl_srv_enable(&light_ctrl_srv);
+
+	/* Publish when transition is completed. */
+	expected_flags = expected_flags & ~BIT(FLAG_TRANSITION);
+	k_sleep(K_MSEC(5)); // Sleep test to allow timeout to run.
+
+	expected_statemachine_cond(true, expected_state, expected_flags);
+
+	/* Light_Off should not change state or flags */
+	bt_mesh_light_ctrl_srv_off(&light_ctrl_srv);
+	expected_statemachine_cond(true, expected_state, expected_flags);
+
+	/* FADE_ON state, Light_On should change state */
```

Comment:
In FADE_ON state, Light_On should **not** change the state? 

The comment is bit confusing to me. The code is about 'entry to FADE_ON state'. Similarly for other comments. It will be nice to fix this for maintainability.